### ISSUE TITLE
Fix timestamps

### ISSUE THREAD

**Issue by obycode** [hirosystems/stacks-subnets#239]
Timestamps are off somewhere, resulting in the explorer showing invalid times: 
<img width="238" alt="image" src="https://user-images.githubusercontent.com/1473715/224459700-1e32ecb6-6503-40c5-aba8-838da5e76ea0.png">

**Comment by jbencin** [comment]
Timestamps are inserted into database in the function `insert_block_snapshot()` in chainstate/burn/db/sortdb.rs. I was able to was able to record the block insertion time using `SystemTime::now()` and read this value via the `/extended/v1/block` API endpoint. This is probably a reasonable approximation and an improvement over `0`, but I think the timestamp we actually want is from the most recent Bitcoin block, since the field is named `burn_block_time` and because Stacks blocks don't seem to keep their own timestamps (I  would guess they are found at the same time as Bitcoin blocks, and so share the same timestamp). I'll keep looking for this...

**Comment by obycode** [comment]
Yup, since it's burn block time, I think that would be right to use the timestamp from the stacks block. It should be available in the `BlockSnapshot` object.

**Comment by jbencin** [comment]
It's available in `BlockSnapshot`, but it's already `0` there. I think the problem is in burnchains/burnchain.rs:

```rust
impl BurnchainBlock {
    pub fn timestamp(&self) -> u64 {
        0
    }

    pub fn header(&self) -> BurnchainBlockHeader {
        match self {
            BurnchainBlock::StacksSubnetBlock(b) => BurnchainBlockHeader {
                block_height: self.block_height(),
                block_hash: self.block_hash(),
                parent_block_hash: self.parent_block_hash(),
                num_txs: b.ops.len() as u64,
                timestamp: self.timestamp(),
            },
        }
    }
}
```

When `BurnchainBlock::header()` is called it will always return a header with `timestamp` set to zero. `BurnchainBlock` seems to have no field for storing a timestamp:

```rust
pub struct StacksSubnetBlock {
    pub current_block: StacksBlockId,
    pub parent_block: StacksBlockId,
    pub block_height: u64,
    pub ops: Vec<StacksSubnetOp>,
}

/// Enum for wrapping Layer-1 blocks for subnets
pub enum BurnchainBlock {
    StacksSubnetBlock(StacksSubnetBlock),
}
```

Also, I'm not sure why `BurnchainBlock` would be represented as a `StacksSubnetBlock`. It looks network messages are parsed into this struct in `sync_with_indexer()`, so I'll see if there is any timestamp available there

**Comment by jbencin** [comment]
I traced this all the way back to `from_new_block_event()` in events.rs. The timestamp was not being extracted from `NewBlock`, and `StacksSubnetBlock` did not have a corresponding timestamp field
